common/utils/formatters.js

```diff
@@ -0,0 +1,76 @@
+import { coerceEmptyStringToUndefined } from 'common/utils/string-utils';
+import { isFilledArray } from 'common/utils/array-utils';
+
+// TODO: Remove eslint disable when more items are exported
+/* eslint-disable import/prefer-default-export */
+
+/**
+ * An object representing the application's visual breakpoints.
+ * @typedef {{
+ * branchOfService: string,
+ * companyName: string,
+ * companyRole: string,
+ * disciplines: string[],
+ * doesWantMentor: boolean,
+ * doesWantScholarshipInfo: boolean,
+ * doesWantToVolunteer: boolean,
+ * employmentStatus: string,
+ * militaryStatus: string,
+ * payGrade: string,
+ * programmingLanguages: string[],
+ * yearsOfService: number,
+ * }} UserData
+ *
+
+/**
+ * @description Format request body for updating user data
+ *
+ * @exports
+ * @param {UserData}
+ * @returns {UserData} non-undefined values from UserData
+ */
+export const formatUserData = ({
+  branchOfService,
+  companyName,
+  companyRole,
+  disciplines,
+  doesWantMentor,
+  doesWantScholarshipInfo,
+  doesWantToVolunteer,
+  employmentStatus,
+  militaryStatus,
+  payGrade,
+  programmingLanguages,
+  yearsOfService,
+}) => {
+  let interests;
+
+  const hasInterests = isFilledArray(disciplines) || isFilledArray(programmingLanguages);
+
+  if (hasInterests) {
+    const interestItems = [...programmingLanguages, ...disciplines];
+
+    // interests could be a string of many comma-separated items, or a single string
+    interests =
+      interestItems.length > 1
+        ? interestItems.join(', ')
+        : coerceEmptyStringToUndefined(interestItems[0]);
```

Comment:
Given that we already know either `disciplines` or `programmingLanguages` is a non-empty array, can `interestItems[0]` ever be an empty string at this point?